### ISSUE TITLE
CONFIG: Add option to disable floppy drive sounds while accessing SD

### ISSUE THREAD

**Issue by MauriceMEGA65** [MEGA65/mega65-core#622]
Currently the floppy drive is being activated while accessing the SD in order to create floppy drive sounds.

I would like to be able to disable the floppy drive being activated while accessing the SD. Using both media makes it hard to distinguish which drive is being read/written by the sound of it. An enable/disable setting in config would be welcome. Perhaps also reducing wear and tear as a bonus. I will make it a future request

**Comment by lydon42** [comment]
Linked to #621

**Comment by MauriceMEGA65** [comment]
Ah, same issue as #518, so most likely duplicate

**Comment by grubi** [comment]
I'd like to support this one. Drive sounds are no problem when loading a game from time to time. But when the Mega65 is used for heavy development, it can quickly become annoying - plus I fear it will wear off the old, used drives more quickly. I have disconnected my disk drive cable internally, after finding out that the SILENT flag in $D6A0 doesn't work.

**Comment by tmilker** [comment]
I'd also like to express my support for this feature. In addition to not being able to determine if a real disk is being accessed, what happens if we have an actual disk in the drive?  I don't know really know how a floppy drive works, could this cause wear & tear on the disk itself too or an unknown/unexpected bug cause it to be accidentally written to or damaged?

**Comment by KiDraDragon** [comment]
It would be a very much appreciated feature.

**Comment by Mikey351AU** [comment]
I support this request.

**Comment by LeifBloomquist** [comment]
Yes!  Need this

**Comment by nobruinfo** [comment]
I would even suggest to make this *opt-in*. Meaning it should not only be configurable but **off** at default. Not only would this make it more VICE compatible :) but also people would not have surprise effects upon first starts of their machines.

**Comment by lydon42** [comment]
This needs #621 to be fixed.

**Comment by deathybrs** [comment]
I add my voice to this request - it is a really cool feature, but many of us fear for the life of our drives, which seems like a valid concern to me.  I further really like @nobruinfo's suggestion that the option default to off rather than on so that we start out protecting people's disk drive.

**Comment by dansanderson** [comment]
Notes from Paul on the VHDL side of this:

* Most likely in sdcardio.vhdl
* the line that enables the real drive motor is motor <= fastio_wdata(5)
* hang on, no, it will be f_motora/f_motorb
* silent_sdcard is the variable that controls it, and it should be working at first look.
* It is not currently applied to head stepping, however.
* so this is likely what is happening: The actual selection of real drive is suppressed until you step tracks, at which point it does get selected.
* basically follow the f_selecta/b references and add the missing checks for real floppy or silent_sdcard to those blocks.

Once we have a working register, it'll be straightforward to add it to Config.

**Comment by nobruinfo** [comment]
@dansanderson Thanks for this list to remember. I was warming it up. Unfortunately I cannot be the one to do the work. Also we need a decision:
* configurable=yes?
* simulation default setting=off?

At least this is what I would wish for with both, unburdened beginners and hard-core retro ears, in mind. ;)

**Comment by lydon42** [comment]
If unsure: also add to ONBOARD, so the user gets asked as soon as possible.

**Comment by RetroCogs** [comment]
As someone that develops late at night, the drive sounds everytime I launch an iteration are very loud and I would love to be able to develop in silence.

**Comment by tundraC** [comment]
I'd love the option to en/disable.  I was troubleshooting a boot issue last year and it was very difficult to figure out what was being used for loading.  The issue was very hard to determine if the drive was actually seeking or faking it.  Plus, when working late at night, it's nice to keep it quiet.  Just my two thoughts :)

**Comment by dansanderson** [comment]
I can confirm that this is implemented and working in the latest development core. This issue can be closed (along with #621).

**Comment by MauriceMEGA65** [comment]
👌
